pkg/tfshim/sdk-v2/provider.go

```diff
@@ -148,7 +148,7 @@ func (p v2Provider) Refresh(
 		return nil, fmt.Errorf("unknown resource %v", t)
 	}
 
-	state, err := upgradeResourceState(ctx, p.tf, r, stateFromShim(s))
+	state, err := upgradeResourceState(ctx, t, p.tf, r, stateFromShim(s))
```

Comment:
Should this use GRPC version? Possibly later?

Reply:
It does. `upgradeResourceState` shims from the representation used by `v2Provider` to `cty.Value`, invokes `upgradeResourceStateGRPC` and then shims back.